### PR TITLE
{RDBMS} `az postgres flexible-server update`: Support Case-Insensitive Input for `--tier`, `--performance-tier`, `--sku`, and `--maintenance-window`

### PR DESCRIPTION
**Pull request by mattboentoro**
**Related command**
az postgres flexible-server update

**Description**
Currently, we have `--high-availability`, `--geo-redundant-backup`, and `--public-access` parameters that supported case-insensitive input. This PR will bring consistency to `--tier`, `--performance-tier`, `--sku`, and `--maintenance-window` parameters.

**Testing Guide**
- Manual testing with 
  - `az postgres flexible-server update --ids <server-id> --sku-name Standard_B16MS`
  - `az postgres flexible-server update --ids <server-id> --performance-tier p15`
  - `az postgres flexible-server update --ids <server-id> --sku-name standard_d4s_v3`
  - `az postgres flexible-server update --ids <server-id> --maintenance-window SUN`
- Added unit test

**History Notes**

[RDBMS] `az postgres flexible-server update`: Support Case-Insensitive Input for `--tier`, `--performance-tier`, `--sku`, and `--maintenance-window`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

**Comment by mattboentoro** [comment]
@evelyn-ys @yonzhan I recently joined PostgreSQL team, and this is my first PR on GitHub for RDBMS. I was wondering why I couldn't get to add my team member as a reviewer? Is there any other permission that I need to obtain for me to do that?

**Comment by mattboentoro** [comment]
@microsoft-github-policy-service agree company="Microsoft"Matthew BoentoroOn Sep 4, 2024, at 10:53, microsoft-github-policy-service[bot] ***@***.***> ***@***.*** agree company="Microsoft"

**Comment by evelyn-ys** [comment]
> @evelyn-ys @yonzhan I recently joined PostgreSQL team, and this is my first PR on GitHub for RDBMS. I was wondering why I couldn't get to add my team member as a reviewer? Is there any other permission that I need to obtain for me to do that?

I noticed that @ambrahma and @nasc17 have already been reviewer. Any other help needed?

**Comment by nasc17** [comment]
> > @evelyn-ys @yonzhan I recently joined PostgreSQL team, and this is my first PR on GitHub for RDBMS. I was wondering why I couldn't get to add my team member as a reviewer? Is there any other permission that I need to obtain for me to do that?
> 
> I noticed that @ambrahma and @nasc17 have already been reviewer. Any other help needed?

I included us in this PR, however the option to add reviewers was not shown for @mattboentoro

File: src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py [hunk 1]
```diff
@@ -209,6 +214,52 @@ def _test_flexible_server_mgmt(self, database_engine, resource_group):
...
+    def _test_flexible_server_mgmt_case_insensitive(self, database_engine, resource_group):
+
+        if self.cli_ctx.local_context.is_on:
+            self.cmd('config param-persist off')
+
+        version = '16'
+        storage_size = 128
+        location = self.postgres_location
+        sku_name = 'Standard_D2s_v3'
+        memory_optimized_sku = 'Standard_E2ds_v4'
```
Comment (nasc17) [inline src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py:226]: nit: Move this to line 241 if only used for update

File: src/azure-cli/azure/cli/command_modules/rdbms/validators.py [hunk 2]
```diff
@@ -313,14 +313,15 @@ def pg_arguments_validator(db_context, location, tier, sku_name, storage_gb, ser
...
     geo_backup_supported = list_location_capability_info['geo_backup_supported']
     _network_arg_validator(subnet, public_access)
     _pg_tier_validator(tier, sku_info)  # need to be validated first
     if tier is None and instance is not None:
-        tier = instance.sku.tier
-    if "supported_storageV2_size" in sku_info[tier]:
-        supported_storageV2_size = sku_info[tier]["supported_storageV2_size"]
+        tier = instance.sku.tier.lower()
+    if "supported_storageV2_size" in sku_info[tier.lower()]:
+        supported_storageV2_size = sku_info[tier.lower()]["supported_storageV2_size"]
```
Comment (nasc17) [inline src/azure-cli/azure/cli/command_modules/rdbms/validators.py:324]: tier is already set to instance.sku.tier.lower(), do you need to set lower again

File: src/azure-cli/azure/cli/command_modules/rdbms/validators.py [hunk 3]
```diff
@@ -367,18 +368,18 @@ def _valid_ssdv2_range(storage_gb, sku_info, tier, iops, throughput, instance):
     storage_throughput = throughput if throughput is not None else instance.storage.throughput
 
     # find min and max values for storage
-    supported_storageV2_size = sku_info[tier]["supported_storageV2_size"]
+    supported_storageV2_size = sku_info[tier.lower()]["supported_storageV2_size"]
```
Comment (nasc17) [inline src/azure-cli/azure/cli/command_modules/rdbms/validators.py:371]: nit: set a variable to sku_tier = tier.lower() to use through method

File: src/azure-cli/azure/cli/command_modules/rdbms/validators.py (hunk 2 above)
Comment (mattboentoro) [reply src/azure-cli/azure/cli/command_modules/rdbms/validators.py:324]: There's a possibility that tier is not in lower case when it is not in the if loop. I'd like to keep this lower case here.

Comment (nasc17) [reply src/azure-cli/azure/cli/command_modules/rdbms/validators.py:324]: Oh I see, tier is an input for the method, and it is also being reset.

**Comment by nasc17** [comment]
@evelyn-ys This PR looks good to go. Could you please review.

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**